app/services/GeoServer.js

```diff
@@ -29,6 +29,66 @@ export async function createChallenge(challenge, successCallback, failureCallbac
   }
 }
 
+
+export async function getChallenges() {
+  let formData = new FormData();
+  formData.append('token', 'geo-ninjas');
+  formData.append('user', 2);  //for now
+
+  try{
+    const response = await fetch('http://enexia.com:10000/geo-challenge/challenge/search', {
+      method: 'POST',
+      body: formData
+    })
+    const responseJson = await response.json();
+    if(responseJson.success) {
+      return responseJson.challenges;
+    } else {
+      return undefined;
+    }
+  } catch (e){
+    failureCallback(e);
```

Comment:
`failureCallback` isn't defined at this point. You need to find some other way to handle exceptions. You could pass in a failureCallback (that's what I did in mine and probably where you got this) or you can just possibly just return undefined. Conceptually I was thinking a toast or snackbar would appear with the error message, but I never got to coding that on my part. Just figure out how you want to handle this.